src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -176,6 +176,225 @@ void ImageReplayer<I>::BootstrapProgressContext::update_progress(
 }
 
 template <typename I>
+void ImageReplayer<I>::ReplayStatusMaker::send(Context *on_finish) {
+  dout(20) << dendl;
+
+  if (m_on_finish) {
+    dout(20) << "previous request is still in progress, ignoring" << dendl;
+    on_finish->complete(-EAGAIN);
+    return;
+  }
+
+  m_on_finish = on_finish;
+  get_registered_clients();
+}
+
+template <typename I>
+void ImageReplayer<I>::ReplayStatusMaker::get_registered_clients() {
```

Comment:
Do you need to query the registered clients?  They should be auto-updating whenever a journal header update occurs.